### PR TITLE
Add index to dbObject

### PR DESCRIPTION
**Pull request by ChrisSchinnerl**

### CODE REVIEW

File: stores/metadata.go [hunk 1]
```diff
@@ -124,6 +124,7 @@ type (
 	dbSlice struct {
```
Comment (ChrisSchinnerl) [inline stores/metadata.go:124]: Reminder to test and merge https://github.com/SiaFoundation/renterd/pull/747 first

**Review by peterjan — APPROVED** [review]
LGTM 👍  - nice find

File: stores/migrations.go [hunk 2]
```diff
@@ -1360,3 +1366,55 @@ func performMigration00030_defaultMigrationSurchargeMultiplier(txn *gorm.DB, log
...
+			UPDATE slices
+			JOIN (
+			    SELECT
+			        id,
+			        ROW_NUMBER() OVER (PARTITION BY db_object_id ORDER BY id) AS new_index
+			    FROM
+			        slices
+				) AS RowNumbered ON slices.id = RowNumbered.id
+			SET
+			    slices.object_index = RowNumbered.new_index;
```
[the remark this answers is not in the document]
Comment (ChrisSchinnerl) [reply stores/migrations.go:1409]: What do you not like about it? I have honestly just copied the slab index query's formatting

File: stores/migrations.go [hunk 3]
```diff
@@ -1360,3 +1366,55 @@ func performMigration00030_defaultMigrationSurchargeMultiplier(txn *gorm.DB, log
...
+	// create column
+	if err := txn.Table("slices").Migrator().AutoMigrate(&struct {
+		ObjectIndex uint `gorm:"index"`
+	}{}); err != nil {
+		return err
+	}
+
+	// populate the column
+	var err error
+	if isSQLite(txn) {
```
[the remark this answers is not in the document]
Comment (ChrisSchinnerl) [reply stores/migrations.go:1382]: SQLite doesn't let you do the UPDATE with JOIN. So the partitioning happened for every call of the subquery and produces wrong results. Whereas in MySQL it only runs once.

**Pull request merged**